packages/shared-ux/page/kibana_template/mocks/src/storybook/inner.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { AbstractStorybookMock } from '@kbn/shared-ux-storybook-mock';
+import type { ArgumentParams } from '@kbn/shared-ux-storybook-mock';
+
+import { NoDataConfigPageStorybookMock } from '@kbn/shared-ux-page-no-data-config-mocks';
+import type { NoDataConfigPageStorybookParams } from '@kbn/shared-ux-page-no-data-config-mocks';
+
+import type {
+  KibanaPageTemplateProps,
+  KibanaPageTemplateServices,
+} from '@kbn/shared-ux-page-kibana-template-types';
+
+type PageHeader = NonNullable<KibanaPageTemplateProps['pageHeader']>;
+export type PageHeaderArguments = Pick<
+  PageHeader,
+  'iconType' | 'pageTitle' | 'description' | 'rightSideItems'
+>;
+
+type PropArguments = Pick<KibanaPageTemplateProps, 'isEmptyState'> & PageHeaderArguments;
+
+export type Params = ArgumentParams<PropArguments, {}> & NoDataConfigPageStorybookParams;
+
+const noDataConfigMock = new NoDataConfigPageStorybookMock();
+
+export const pageHeaderArguments: ArgumentParams<PropArguments> = {
+  isEmptyState: {
+    control: 'boolean',
+    defaultValue: false,
+  },
+  iconType: {
+    control: { type: 'radio' },
+    options: ['logoElastic', 'logoKibana', 'logoCloud', undefined],
+    defaultValue: undefined,
+  },
+  pageTitle: {
+    control: 'text',
+    defaultValue: 'Page title',
+  },
+  description: {
+    control: 'text',
+    defaultValue: 'Page description',
+  },
+  rightSideItems: {
+    control: 'boolean',
+    defaultValue: true,
+  },
+};
+
+const rightSideItems: PageHeaderArguments['rightSideItems'] = [
+  <span>First Item</span>,
+  <span>Second Item</span>,
+];
+
+export class StorybookMock extends AbstractStorybookMock<
```

Comment:
This feels a bit overly complicated. Do we need this `AbstractStorybookMock`?

Reply:
I needed a way to make sure that all Storybook mocks followed the same API.  This allows us to compose mocked values for services without having to know what services consumed components need.